src/test/java/org/jenkinsci/plugins/saml/LiveTest.java

```diff
@@ -0,0 +1,215 @@
+
+        GlobalSecurityConfig sc = new GlobalSecurityConfig(jenkins);
+        sc.open();
+
+        // Authentication
+        SamlSecurityRealm realm = configureBasicSettings(sc);
+        String idpMetadata = readIdPMetadataFromURL(samlServer);
+        realm.setXml(idpMetadata);
+        realm.setBinding(SAML2_POST_BINDING_URI);
+        configureEncrytion(realm);
+        configureAuthorization(sc);
+
+        waitFor().withTimeout(10, TimeUnit.SECONDS).until(() -> hasContent("Enter your username and password")); // SAML service login page
+
+        // SAML server login
+        makeLoginWithUser1();
+    }
+
+    @Test
+    public void authenticationFail() throws IOException, InterruptedException {
+        jenkins.open(); // navigate to root
+        String rootUrl = jenkins.getCurrentUrl();
+        SAMLContainer samlServer = startSimpleSAML(rootUrl);
+
+        GlobalSecurityConfig sc = new GlobalSecurityConfig(jenkins);
+        sc.open();
+
+        // Authentication
+        SamlSecurityRealm realm = configureBasicSettings(sc);
+        String idpMetadata = readIdPMetadataFromURL(samlServer);
+        realm.setXml(idpMetadata);
+
+        configureEncrytion(realm);
+        configureAuthorization(sc);
+
+        waitFor().withTimeout(10, TimeUnit.SECONDS).until(() -> hasContent("Enter your username and password")); // SAML service login page
+
+        // SAML server login
+        find(by.id("username")).sendKeys("user1");
+        find(by.id("password")).sendKeys("WrOnGpAsSwOrD");
+        find(by.button("Login")).click();
+
+        waitFor().withTimeout(5, TimeUnit.SECONDS).until(() -> hasContent("Either no user with the given username could be found, or the password you gave was wrong").matchesSafely(driver)); // wait for the login to propagate
+        assertThat(jenkins.getCurrentUrl(), containsString("simplesaml/module.php/core/loginuserpass.php"));
+    }
+    */
+
+    private String readIdPMetadataFromURL() throws IOException {
+        // get saml metadata from IdP
+        URL metadata = new URL(createIdPMetadataURL());
+        URLConnection connection = metadata.openConnection();
+        return IOUtils.toString(connection.getInputStream(), StandardCharsets.UTF_8);
+    }
+
+    private String createIdPMetadataURL() {
+        return "http://" + samlContainer.getHost() + ":" + samlContainer.getFirstMappedPort() + "/simplesaml/saml2/idp/metadata.php";
+    }
+
+    private static void configureAuthorization() {
+        Jenkins.get().setAuthorizationStrategy(new MockAuthorizationStrategy().
+                grant(Jenkins.ADMINISTER).everywhere().to("group1").
+                grant(Jenkins.READ).everywhere().to("group2"));
+    }
+
+    private static SamlSecurityRealm configureBasicSettings(IdpMetadataConfiguration idpMetadataConfiguration, SamlAdvancedConfiguration advancedConfiguration) throws IOException {
+        // TODO use @DataBoundSetter wherever possible and load defaults from DescriptorImpl
```

Comment:
https://www.jenkins.io/doc/developer/plugin-development/pipeline-integration/#constructor-vs-setters

Reply:
(Would likely also be better for CasC: should be able to omit default values of attributes.)